internal/gh/rest.go

```diff
@@ -0,0 +1,164 @@
+package gh
+
+import (
+	"net/http"
+    "regexp"
+    "log/slog"
+    "encoding/json"
+)
+
+const baseURL = "https://api.github.com"
+var NumberForcePush = 0
+
+type Result struct {
+    Entry string
+}
+
+func createHttpRequest(client *http.Client, url string, token string, queryParameters map[string] string)(*http.Request, error) {
+/*
+    Create an HTTP request using the parameters, token and http client.
+*/
+
+    req, err := http.NewRequest("GET", url, nil)
+    if err != nil {
+        return nil, err
+    }
+        
+    req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+token)
+    
+    query := req.URL.Query()
+
+    for key, value := range queryParameters{
+        query.Add(key, value)
+    }
+    req.URL.RawQuery = query.Encode()
+
+    return req, nil
+}
+
+func GetForcePushInfo(owner string, repo string, token string, branch string) (int, error){
+/*
+    Entry point for the Force Push processing, setting the url, query paramenters and creating the client.
+*/
+    slog.Default().Info("Getting repo info - getForcePushInfo method")
+    repoActivityUrl := baseURL + "/repos/" + owner + "/" + repo + "/activity"
+    queryParameters := map[string]string{
+        "per_page" : "100",
+        "activity_type" : "force_push",
+        "ref" : branch,
+    }
+
+    client := &http.Client{}
+
+    err := processForcePushRequest(client, repoActivityUrl, token, queryParameters) 
+    if err != nil {
+        slog.Default().Info("Getting repo info falied - %v getForcePushInfo method", err)
+        return 0, err
+    }    
+    return NumberForcePush, nil
+}
+
+func executeHTTPRequest(client *http.Client, req *http.Request) (*http.Response, error){
+/*
+    Execute the HTTP request and return response
+*/
+
+    resp, err := client.Do(req)
+	RequestCounter++
+	if err != nil {
+        return nil, err
+	}
+    return resp, nil
+}
+
+func processForcePushRequest(client *http.Client, repoActivityUrl string, token string, queryParameters map[string] string) error {
+/*
+    Method where calls to other methods (createHTTPRequest, executeHTTPRequest, processHttpResponse) is handled and loop is added so that requests are processed till there no rel = "next"
+*/
+
+    slog.Default().Info("Getting Force Push Info - processForcePushRequestAndResponse method")
+    
+    for true{
+        httpReq, err := createHttpRequest(client, repoActivityUrl, token, queryParameters)
```

Comment:
I'm pretty sure that you can reuse the HTTP Request once it has been created. you just need to update the query parameters of the struct that is created by your method.